File: C++/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        
        for (int n: nums) {
            freq[n]++;
        }
        
        vector<int> counts;
        
        for (auto it: freq) {
            counts.push_back(it.second);
        }
        
        sort(counts.begin(), counts.end());
        
        vector<int> ans;

        for (auto it: freq) {
            for (int i = 0; i < k; i++) {
                int c = counts[counts.size() - i - 1];
                if (it.second == c) {
                    ans.push_back(it.first);
                    break;
                }
            }
        }
        
        
        return ans;
    }
};

File: JavaScript/top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    const hist = {}
    for (const n of nums) {
        hist[n] = (hist[n] || 0) + 1
    }
    const ans = []
    while (k > 0) {
        let maxFreq = -1
        let maxElem = null
        for (const [elem, freq] of Object.entries(hist)) {
            if (maxFreq < freq) {
                maxFreq = freq
                maxElem = elem
            }
        }
        ans.push(maxElem)
        hist[maxElem] = 0
        k--
    }
    return ans
};